Exercises.IEnumerable2/Item.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Exercises.IEnumerable2.Dependences;
+using System.Collections.Generic;
+
+namespace Exercises.IEnumerable2
+{
+    public abstract class Item
+    {
+        private const string Name = "Item";
+
+
+        public IList<Item> ChildItems { get; private set; }
+
+        public IEnumerable<Item> ItemsInDepth => new ItemInDepthCollection(this);
+
+        public IEnumerable<Item> ItemInWidth => new ItemInWidthCollection(this);
```

Comment:
Having property which creates new Item on each call might be misleading for those who use your class. Overall property looks like a field, and having `Object.ReferenceEqual(myClass.SomeProp, myClass.SomeProp) == false` might rise a lot of questions